eth/stagedsync/stage_polygon_sync.go

```diff
@@ -267,29 +267,51 @@ func UnwindEvents(tx kv.RwTx, unwindPoint uint64) error {
 	if err != nil {
 		return err
 	}
-
 	defer cursor.Close()
+
 	var blockNumBuf [8]byte
 	binary.BigEndian.PutUint64(blockNumBuf[:], unwindPoint+1)
-	k, eventId, err := cursor.Seek(blockNumBuf[:])
+
+	_, _, err = cursor.Seek(blockNumBuf[:])
 	if err != nil {
 		return err
 	}
-	if k != nil {
-		eventCursor, err := tx.RwCursor(kv.BorEvents)
-		if err != nil {
-			return err
-		}
-		defer eventCursor.Close()
-		for eventId, _, err = eventCursor.Seek(eventId); err == nil && eventId != nil; eventId, _, err = eventCursor.Next() {
-			if err = eventCursor.DeleteCurrent(); err != nil {
-				return err
-			}
-		}
-		if err != nil {
+
+	_, prevSprintLastIDBytes, err := cursor.Prev() // last event ID of previous sprint
+	if err != nil {
+		return err
+	}
+
+	var prevSprintLastID uint64
+	if prevSprintLastIDBytes == nil {
+		// we are unwinding the first entry, remove all items from BorEvents
+		prevSprintLastID = 0
+	} else {
+		prevSprintLastID = binary.BigEndian.Uint64(prevSprintLastIDBytes)
+	}
+
+	eventId := make([]byte, 8) // first event ID for this sprint
+	binary.BigEndian.PutUint64(eventId[:], prevSprintLastID+1)
+
+	eventCursor, err := tx.RwCursor(kv.BorEvents)
+	if err != nil {
+		return err
+	}
+	defer eventCursor.Close()
+
+	for eventId, _, err = eventCursor.Seek(eventId); err == nil && eventId != nil; eventId, _, err = eventCursor.Next() {
+		if err = eventCursor.DeleteCurrent(); err != nil {
 			return err
 		}
 	}
+	if err != nil {
+		return err
+	}
+
+	k, _, err := cursor.Seek(blockNumBuf[:]) // move cursor back to this sprint
```

Comment:
instead of doing another Seek could this be cursor.Next()?

Reply:
Is there a performance benefit to this? I originally had it as `Next` but I felt the code for `Seek` was cleaner.

Reply:
Seek is O(Log(N)) afaik - mdbx is a tree of pages
Next is O(1)